### ISSUE TITLE
More comments

### ISSUE THREAD

**Issue by lectricas** [forceLain/AwesomeRecyclerView#2]
Hi.
LayoutManager is a very complicated stuff for me, but I really want to know how to create one.
Can you supply your code with more comments to help other programmers.

**Comment by forceLain** [comment]
You can check out my article (in russian) : https://habrahabr.ru/company/eastbanctech/blog/267497/
This project just a sample code for the article

**Comment by lectricas** [comment]
Yeah, I've read that. And the first queston that I have for example is:

`public class AwesomeLayoutManager extends RecyclerView.LayoutManager {
    @Override
    public RecyclerView.LayoutParams generateDefaultLayoutParams() {
        return new RecyclerView.LayoutParams(RecyclerView.LayoutParams.MATCH_PARENT, RecyclerView.LayoutParams.MATCH_PARENT);
    }
}`

and you are saying:

_Lets implement  generateDefaultLayoutParams() which returns suitable LayoutParams for views, which LayoutParams don't satisfy us._
And the questions are: 
1)are this "views" is our child views? Which is set in Adapter as article_item.xml?
2)Why they do not satisfy as?

Then for example:

measureChildWithMargins(view, 0, 0);  why we need to measure it?

So I stack in the beginning, imagine how tough for me other 500 lines of code :D

**Comment by forceLain** [comment]
> 1)are this "views" is our child views? Which is set in Adapter as article_item.xml?

Yes, views are provided by our adapter. Adapter may give us various views: view may have RecyclerView.LayoutParams, SomeOtherViewGroup.LayoutParams or may have no parameters at all. If the LayoutParams of the provided view are not RecyclerView.LayoutParams or null, then RecyclerView ask us to create suitable LayoutParams for the view.

> 2)Why they do not satisfy as?

Any RecyclerView.LayoutParams are satisfy us. Any Params which are not RecyclerView.LayoutParams do not satisfy as.

> measureChildWithMargins(view, 0, 0); why we need to measure it?

Not sure I correctly understand the question, we should call measure for every child. It's part of standard Android measure/layout/draw mechanism https://developer.android.com/guide/topics/ui/how-android-draws.html (except that RV draws children itself)
